pkg/utils/api_discover.go

```diff
@@ -0,0 +1,53 @@
+package utils
+
+import (
+	"fmt"
+	nativelog "log"
+	"strings"
+
+	datav1alpha1 "github.com/fluid-cloudnative/fluid/api/v1alpha1"
+	"github.com/fluid-cloudnative/fluid/pkg/utils/testutil"
+	"github.com/pkg/errors"
+	"k8s.io/client-go/discovery"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+var enabledFluidResources map[string]bool = nil
+
+func init() {
+	if testutil.IsUnitTest() {
+		return
+	}
+	discoverFluidResourcesInCluster()
+	allEnabledResources := []string{}
+	for resource := range enabledFluidResources {
+		allEnabledResources = append(allEnabledResources, resource)
+	}
+
+	nativelog.Printf("Discovered Fluid CRDs in cluster: %v, enable related reconcilers only", allEnabledResources)
```

Comment:
Can you use `log.Info` ? Because I saw `log` is initialized in pkg/utils/yaml.go.

Reply:
Golang doesn't guarantee the execution order of multiple `func init` in a same package. So it is possible that the log  is not initialized when `func init` starts in `api_discover.go`